OpenRA.Mods.Common/Widgets/WorldCommandWidget.cs

```diff
@@ -191,15 +191,28 @@ bool CycleBases()
 				.Select(b => b.Actor)
 				.ToList();
 
-			if (!bases.Any())
-				return true;
+            var noBase = false;
+
+            // If no base found the nondestroyed buildings are selected.
+            if (!bases.Any())
+            {
+                bases = world.ActorsWithTrait<TargetableBuilding>()
```

Comment:
We don't want to introduce edge cases here if a custom map assigns a non selectable building to a player.